compiler/src/test/java/com/squareup/anvil/compiler/codegen/ContributesBindingGeneratorTest.kt

```diff
@@ -309,6 +309,22 @@ class ContributesBindingGeneratorTest {
     }
   }
 
+  @Test fun `the contributed binding class can extend Any explicitly`() {
+    compile(
+      """
+      package com.squareup.test
+
+      import com.squareup.anvil.annotations.ContributesBinding
+
+      @ContributesBinding(Int::class, boundType = Any::class)
+      interface ContributingInterface
+      """
+    ) {
+      assertThat(contributingInterface.hintBinding?.java).isEqualTo(contributingInterface)
```

Comment:
Is this assert the one that's checking for `Any`? This one isn't clear to me. E.g. why are we checking `hintBinding?.java` instead of just `hintBinding`, and why are we checking if it's equal to `contributingInterface` instead of `Any::class`

Reply:
A hint is always the type that is contributed and the scope. Based on that information Anvil can fetch the annotation and read all values. `hintBinding` returns a `KClass` and `contributingInterface` is a `Class`, so one or the other has to be converted.